azurerm/internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -707,3 +720,43 @@ func resourceKubernetesClusterNodePoolDelete(d *schema.ResourceData, meta interf
 
 	return nil
 }
+
+func upgradeSettingsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"max_surge": {
+					Type:     schema.TypeString,
+					Optional: true,
```

Comment:
since this is the only field within this block - which is Optional - we can make this inner field Required:

```suggestion
					Required: true,
```

this means if someone wants to specify the `upgrade_settings` block they'll need to specify the inner value, else omit the `upgrade_settings` block